### PR TITLE
CICD: Add APK Signing for the Android CICD Pipeline.

### PR DESCRIPTION
**Pull request by mikeller**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Add signing of the android APK to the scripts used by the CICD pipeline.
Also update the example for running these scripts locally, and add the artefacts generated by doing so to .gitignore.

### Changes made:

1. add signing of the android APK to the scripts used by the CICD pipeline;
2. update the pipeline .yml;
3. update the example for running these scripts locally;
4. add the artefacts generated by doing so to .gitignore.

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -93,10 +96,14 @@ jobs:
           This build is based on http://github.com/subsurface/subsurface/commit/${{ github.sha }}
 
           None of these artifacts are signed.
-          The Android APK can be side-loaded on most Android devices. If you had a previous Subsurface-mobile version installed from the Google Play store, you'll have to uninstall that first.
+          The Android APK can be side-loaded on most Android devices.
```
Comment (dirkhh) [inline .github/workflows/android.yml]: So that assumes that I add the existing keystore / password as secrets, correct?

**Review by dirkhh — COMMENTED** [review]
This looks like it's going in the right direction...

File: .github/workflows/android.yml (hunk 1 above)
Comment (mikeller) [reply .github/workflows/android.yml]: I sent you an email about that. I'd probably generate a new key and keystore dedicated to CICD releases - there doesn't seem to be a downside to using self signed keys for this.

Comment (glance-) [reply .github/workflows/android.yml]: The android verification process is basically that on upgrade, the new package is signed by the same key that the old package is signed by. That way you can't switch from one signing lineage of package to another signing lineage without uninstalling the app and removing the data.

Comment (dirkhh) [reply .github/workflows/android.yml]: Correct - and so our CICD binaries aren't using the same key as the Google Play store binaries (I was convinced that that would be a violation of the terms of my developer agreement), which forces this first reinstall. But after that, every CICD build uses the same key and therefore upgrades work.

**Pull request merged**